messaging/src/main/java/org/axonframework/eventhandling/TerminalEventMessage.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling;
+
+import org.axonframework.messaging.MessageType;
+
+/**
+ * Empty {@link EventMessage} implementation without any {@link EventMessage#getPayload() payload}, used as the
+ * <b>terminal</b> message of a {@link org.axonframework.messaging.MessageStream}.
+ * <p>
+ * Only useful to be paired with {@link org.axonframework.messaging.Context} information in an event-specific
+ * {@code MessageStream} when there is no event payload to combine it with.
+ *
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+public class TerminalEventMessage extends GenericEventMessage<Void> implements EventMessage<Void> {
+
+    /**
+     * The sole instance of the {@link TerminalEventMessage}.
+     */
+    public static final TerminalEventMessage INSTANCE = new TerminalEventMessage();
+
+    private TerminalEventMessage() {
+        //noinspection DataFlowIssue
+        super(new MessageType(TerminalEventMessage.class), null);
```

Comment:
Interesting, the payload of an event can't be null? Why not? :thinking: 

I recognize it's a weird use-case, but maybe just the occurrence of an event is enough for some domains

Reply:
I made the `EventMessage#payload` non-null when I was doing the `QualifiedName`/`MessageType` introduction.

My frame of mind here was, that I can't imagine why somebody would append an event with nothing in it into the store.